app/controllers/roles_controller.rb

```diff
@@ -52,6 +52,8 @@ def edit
   end
 
   def update
+    @role.set_taxonomies(taxonomy_params!)
```

Comment:
I see couple of reasons, for most of them please see above, one more - since role#set_taxonomies also touches filters and their relations it would be much better to have some kind of service object that does it, something like [composer](https://github.com/theforeman/foreman_remote_execution/blob/master/app/models/job_invocation_composer.rb), definitely not a single model which callbacks rule the world